tcms/issuetracker/azure_boards.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+from requests.auth import HTTPBasicAuth
+import requests
+
+from tcms.core.contrib.linkreference.models import LinkReference
+from tcms.issuetracker.base import IssueTrackerType, IntegrationThread
+
+
+class AzureAPI():
+    """
+        Azure Boards API interaction class.
+
+    """
+    def __init__(self, base_url=None, password=None):
+        self.api_version = "?api-version=6.0"
+        self.headers = {"Accept": 'application/json-patch+json',
+                        'Content-type': 'application/json-patch+json'}
+        self.auth = HTTPBasicAuth('apikey', password)
+        self.base_url = base_url + "/_apis/"
+
+    def get_issue(self, issue_id):
+        url = "{0}{1}{2}{3}".format(self.base_url, "wit/workitems/",
+                                    str(issue_id), self.api_version)
+        return self._get(url, self.headers, self.auth)
+
+    def create_issue(self, body):
+        url = "{0}{1}{2}".format(self.base_url, "wit/workitems/$Issue", self.api_version)
+        return self._post(url, self.headers, self.auth, body)
+
+    def update_issue(self, issue_id, body):
+        url = "{0}{1}{2}{3}".format(self.base_url, "wit/workitems/",
+                                    str(issue_id), self.api_version)
+        return self._patch(url, self.headers, self.auth, body)
+
+    def add_comment(self, issue_id, body):
+        headers = {"Content-type": "application/json"}
+        url = "{0}{1}{2}{3}{4}{5}".format(self.base_url, "wit/workitems/",
+                                          str(issue_id), "/comments", self.api_version,
+                                          "-preview.3")
+        return self._post(url, headers, self.auth, body)
+
+    def get_comments(self, issue_id):
+        headers = {"Content-type": "application/json"}
+        url = "{0}{1}{2}{3}{4}{5}".format(self.base_url, "wit/workitems/",
+                                          str(issue_id), "/comments", self.api_version,
+                                          "-preview.3")
+        return self._get(url, headers, self.auth)
+
+    @staticmethod
+    def _get(url, headers, auth):
+        response = requests.get(url, headers=headers, auth=auth)
+        return response.json()
+
+    @staticmethod
+    def _post(url, headers, auth, body):
```

Comment:
_post and _patch methods can be refactored code climate raises an issue for these two but IMO it's much more readable and clean.

Reply:
doesn't really make a difference but if CodeClimate complains let's make it happy.